### PR TITLE
Validate decimal256 with i256 directly

### LINKED ISSUE #3024: Compare i256 in validate_decimal256_precision
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

`validate_decimal256_precision` validates decimal 256 values by comparing little-endian bytes of decimal value with const byte values. As we moved to i256 now, we can compare i256 directly.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #3024.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-buffer/src/bigint.rs [hunk 1]
```diff
@@ -87,6 +87,15 @@ impl i256 {
         }
     }
 
+    /// Create an integer value from its representation as high and low byte arrays in little-endian.
```
Comment (tustvold) [inline arrow-buffer/src/bigint.rs:90]: Could we just use the from_le_bytes above?

Comment (viirya) [reply arrow-buffer/src/bigint.rs:90]: I originally wanted that. But I cannot make `from_le_bytes` as const fn. 

```
error[E0015]: cannot call non-const fn `Result::<[u8; 16], TryFromSliceError>::unwrap` in constant functions
```

Comment (tustvold) [reply arrow-buffer/src/bigint.rs:90]: https://github.com/apache/arrow-rs/pull/3026

**Review by tustvold — APPROVED** [review]
Love it 👍

FYI @liukun4515

**Comment by viirya** [comment]
Thanks @tustvold

**Pull request merged**